public/app/features/alerting/unified/components/rules/deleted-rules/ConfirmRestoreDeletedRuleModal.tsx

```diff
@@ -0,0 +1,165 @@
+import { backendSrv } from 'app/core/services/backend_srv';
+import { getMessageFromError } from 'app/core/utils/errors';
+import { useAsync } from 'app/features/alerting/unified/hooks/useAsync';
+import { GrafanaRuleDefinition, RulerGrafanaRuleDTO, RulerRuleDTO } from 'app/types/unified-alerting-dto';
+
+import { useAddRuleToRuleGroup } from '../../../hooks/ruleGroup/useUpsertRuleFromRuleGroup';
+import { RuleFormValues } from '../../../types/rule-form';
+import { GRAFANA_RULES_SOURCE_NAME } from '../../../utils/datasource';
+import { stringifyErrorLike } from '../../../utils/misc';
+import { grafanaRuleDtoToFormValues } from '../../../utils/rule-form';
+import { rulerRuleType } from '../../../utils/rules';
+import { createRelativeUrl } from '../../../utils/url';
+
+type ModalProps = Pick<ComponentProps<typeof ConfirmModal>, 'isOpen' | 'onDismiss'> & {
+  isOpen: boolean;
+  ruleToRestore?: RulerGrafanaRuleDTO | undefined;
+  onRestoreSucess: () => void;
+  onRestoreError: (error: Error) => void;
+};
+
+export const ConfirmRestoreDeletedRuleModal = ({
+  isOpen,
+  ruleToRestore,
+  onDismiss,
+  onRestoreSucess,
+  onRestoreError,
+}: ModalProps) => {
+  const [restoreMethod, { error }] = useRestoreDeletedRule();
+  const title = t('alerting.deletedRules.restore-modal.title', 'Restore deleted alert rule');
+  const errorTitle = t('alerting.deletedRules.restore-modal.error', 'Could not restore deleted alert rule');
+  const confirmText = !error
+    ? t('alerting.deletedRules.restore-modal.confirm', 'Yes, restore deleted rule')
+    : 'Manually restore the rule';
+
+  const styles = useStyles2(getStyles);
+
+  async function onRestoreConfirm() {
+    if (!ruleToRestore) {
+      return;
+    }
+    return restoreMethod
+      .execute(ruleToRestore)
+      .then(() => {
+        onDismiss();
+        onRestoreSucess();
+      })
+      .catch((err) => {
+        onRestoreError(err);
+      });
+  }
+
+  async function onManualRestore() {
+    if (!ruleToRestore) {
+      return;
+    }
+    await redirectToRestoreForm(ruleToRestore);
+  }
+
+  return (
+    <ConfirmModal
+      isOpen={isOpen}
+      title={title}
+      confirmText={confirmText}
+      modalClass={styles.modal}
+      confirmButtonVariant={!error ? 'destructive' : 'primary'}
+      body={
+        <Stack direction="column" gap={2}>
+          <Trans i18nKey="alerting.deletedRules.restore-modal.body">
+            Are you sure you want to restore this deleted alert rule definition?
+          </Trans>
+
+          <div>{ruleToRestore && <RulePeview rule={ruleToRestore} />}</div>
+          {error && (
+            <Alert severity="warning" title={errorTitle}>
+              <Trans i18nKey="alerting.alertVersionHistory.restore-deleted-manually">
+                Your alert rule could not be restored. This may be due to changes to other entities such as contact
+                points, data sources etc. Please manually restore the deleted rule.
+              </Trans>
```

Comment:
The message might not be clear - `Manually restore` sounds quite technical and can be confusing. We might need to give the user specific instructions on what they need to do